homeassistant/components/rainbird/switch.py

```diff
@@ -41,124 +39,72 @@
 )
 
 
-async def async_setup_platform(
+async def async_setup_entry(
     hass: HomeAssistant,
-    config: ConfigType,
+    config_entry: ConfigEntry,
     async_add_entities: AddEntitiesCallback,
-    discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
-    """Set up Rain Bird switches over a Rain Bird controller."""
-
-    if discovery_info is None:
-        return
-
-    controller: AsyncRainbirdController = discovery_info[RAINBIRD_CONTROLLER]
-    try:
-        available_stations: AvailableStations = (
-            await controller.get_available_stations()
+    """Set up entry for a Rain Bird irrigation switches."""
+    coordinator = hass.data[DOMAIN][config_entry.entry_id]
+    async_add_entities(
+        RainBirdSwitch(
+            coordinator,
+            zone,
+            config_entry.options.get(ATTR_DURATION, DEFAULT_TRIGGER_TIME_MINUTES),
```

Comment:
We know that the options will contain a default already.
```suggestion
            config_entry.options[ATTR_DURATION],
```